enumerable_method.rb

```diff
@@ -123,38 +151,64 @@ def my_map
     return to_enum(:my_map) unless block_given?
 
     newarray = []
+    if self.class == Array
+      my_each do |x|
+        newarray << if yield x
+                      true
+                    else
+                      false
+                    end
+      end
```

Comment:
you can refactor the code in the following way. That you have already done on the next condition.
```ruby
newarray << yield(x) ? true : false
```

Reply:
thank about your suggestion it works fine for me